homeassistant/components/squeezebox_player/media_player.py

```diff
@@ -0,0 +1,218 @@
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return unique id for entity."""
+        return self.player.player_id
+
+    @property
+    def available(self) -> bool:
+        """Return availability of entity."""
+        return self.player.connected
+
+    @property
+    def state(self) -> str:
+        """Return current state."""
+        if not self.player.powered:
+            return STATE_OFF
+        return STATE_MAPPING[self.player.state]
+
+    @callback
+    def update_attributes(self) -> None:
+        """Handle player updates."""
+        self._attr_name = self.player.name
+        self._attr_volume_level = self.player.volume_level / 100
+        self._attr_media_position = self.player.elapsed_seconds
+        self._attr_media_position_updated_at = utcnow()
+        self._attr_media_content_id = self.player.current_url
+        self._attr_media_content_type = "music"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.player.player_id)},
+            manufacturer=DEFAULT_NAME,
+            model=self.player.device_model or self.player.device_type,
+            name=self.player.name,
+            hw_version=self.player.firmware,
+        )
+        # PiCore player has web interface
+        if "-pCP" in self.player.firmware:
+            self._attr_device_info[
+                "configuration_url"
+            ] = f"http://{self.player.device_address}"
+
+    async def async_media_play(self) -> None:
+        """Send play command to device."""
+        await self.player.play()
+
+    async def async_media_pause(self) -> None:
+        """Send pause command to device."""
+        await self.player.pause()
+
+    async def async_media_stop(self) -> None:
+        """Send stop command to device."""
+        await self.player.stop()
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
+        await self.player.volume_set(volume)
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        await self.player.mute(mute)
+
+    async def async_turn_on(self) -> None:
+        """Turn on device."""
+        await self.player.power(True)
+
+    async def async_turn_off(self) -> None:
+        """Turn off device."""
+        await self.player.power(False)
+
+    async def async_play_media(self, media_type: str, media_id: str, **kwargs) -> None:
+        """Send the play_media command to the media player."""
+        # Handle media_source
+        if media_source.is_media_source_id(media_id):
+            sourced_media = await media_source.async_resolve_media(self.hass, media_id)
+            media_id = sourced_media.url
+            media_type = sourced_media.mime_type
+
+        if not media_type.startswith("audio/"):
+            media_type = None  # type: ignore[assignment]
+        await self.player.play_url(media_id, mime_type=media_type)
```

Comment:
Make sure to prefix the URLs with HA URL if it's relative.

```suggestion
        async_process_play_media_url(self.hass, media_id)

        await self.player.play_url(media_id, mime_type=media_type)
```